diff --git a/server/entity/Media.ts b/server/entity/Media.ts
index 6b418de3..0b0c0b70 100644
--- a/server/entity/Media.ts
+++ b/server/entity/Media.ts
@@ -1,6 +1,10 @@
 import RadarrAPI from '@server/api/servarr/radarr';
 import SonarrAPI from '@server/api/servarr/sonarr';
-import { MediaStatus, MediaType } from '@server/constants/media';
+import {
+  MediaRequestStatus,
+  MediaStatus,
+  MediaType,
+} from '@server/constants/media';
 import { MediaServerType } from '@server/constants/server';
 import { getRepository } from '@server/datasource';
 import { Blocklist } from '@server/entity/Blocklist';
@@ -50,7 +54,18 @@ class Media {
           'media.id= watchlist.media and watchlist.requestedBy = :userId',
           { userId: user?.id }
         )
-        .leftJoinAndSelect('media.requests', 'requests')
+        .leftJoin(
+          'media.requests',
+          'requests',
+          'requests.status NOT IN (:...inactiveStatuses)',
+          {
+            inactiveStatuses: [
+              MediaRequestStatus.DECLINED,
+              MediaRequestStatus.COMPLETED,
+            ],
+          }
+        )
+        .addSelect(['requests.id', 'requests.status'])
         .where(' media.tmdbId in (:...finalIds)', { finalIds })
         .getMany();
 
diff --git a/src/components/MediaSlider/index.tsx b/src/components/MediaSlider/index.tsx
index e7e09cbf..1b6423a5 100644
--- a/src/components/MediaSlider/index.tsx
+++ b/src/components/MediaSlider/index.tsx
@@ -89,10 +89,8 @@ const MediaSlider = ({
       }
 
       return (
-        i.mediaInfo?.status === MediaStatus.AVAILABLE ||
-        i.mediaInfo?.status === MediaStatus.PARTIALLY_AVAILABLE ||
-        !i.mediaInfo?.requests ||
-        i.mediaInfo.requests.length === 0
+        i.mediaInfo?.status !== MediaStatus.PENDING &&
+        i.mediaInfo?.status !== MediaStatus.PROCESSING
       );
     });
   }
diff --git a/src/hooks/useDiscover.ts b/src/hooks/useDiscover.ts
index 95bc9c45..9805fc78 100644
--- a/src/hooks/useDiscover.ts
+++ b/src/hooks/useDiscover.ts
@@ -147,10 +147,8 @@ const useDiscover = <
     titles = titles.filter(
       (i) =>
         (i.mediaType === 'movie' || i.mediaType === 'tv') &&
-        (i.mediaInfo?.status === MediaStatus.AVAILABLE ||
-          i.mediaInfo?.status === MediaStatus.PARTIALLY_AVAILABLE ||
-          !i.mediaInfo?.requests ||
-          i.mediaInfo.requests.length === 0)
+        i.mediaInfo?.status !== MediaStatus.PENDING &&
+        i.mediaInfo?.status !== MediaStatus.PROCESSING
     );
   }
 
